src/gennylib/src/encryption.cpp

```diff
@@ -0,0 +1,527 @@
+// Copyright 2022-present MongoDB Inc.
```

Comment:
Before digging into this, would it make sense to split this into two PRs, one for this functionality and one that integrates it with the rest of Genny?

Reply:
(Also saw there's now comments on this file, so feel free to ignore this point!)